internal/service/advancedcluster/model_sdk_version_conversion.go

```diff
@@ -0,0 +1,130 @@
+package advancedcluster
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	admin20231115 "go.mongodb.org/atlas-sdk/v20231115014/admin"
+	"go.mongodb.org/atlas-sdk/v20240530001/admin"
+)
+
+// Conversions from one SDK model version to another are used to avoid duplicating our flatten/expand conversion functions.
+// - These functions must not contain any business logic.
+// - All will be removed once we rely on a single API version.
+
+func convertTagsToLatest(tags *[]admin20231115.ResourceTag) *[]admin.ResourceTag {
+	if tags == nil {
+		return nil
+	}
+	tagSlice := *tags
+	results := make([]admin.ResourceTag, len(tagSlice))
+	for i := range len(tagSlice) {
+		tag := tagSlice[i]
+		results[i] = admin.ResourceTag{
+			Key:   tag.Key,
+			Value: tag.Value,
+		}
+	}
+	return &results
+}
+
+func convertBiConnectToOldSDK(biconnector *admin.BiConnector) *admin20231115.BiConnector {
+	if biconnector == nil {
+		return nil
+	}
+	return &admin20231115.BiConnector{
+		Enabled:        biconnector.Enabled,
+		ReadPreference: biconnector.ReadPreference,
+	}
+}
+
+func convertLabelSliceToOldSDK(slice []admin.ComponentLabel, err diag.Diagnostics) ([]admin20231115.ComponentLabel, diag.Diagnostics) {
+	if err != nil {
```

Comment:
seems strange to have err param just to check if it's set. i suppose it's a convenience for the caller

Reply:
correct, essentially enables this code from call perspective:
```
labels, err := convertLabelSliceToOldSDK(expandLabelSliceFromSetSchema(d))
if err != nil {
	return err
}
cluster.Labels = &labels